### PR TITLE
Compat python 3.8

### LINKED ISSUE #3776: asyncio.coroutine is deprecated in 3.8
`asyncio.coroutine` decorator is deprecated with 3.8 and is used in some of the code. DeprecationWarnings could be fixed.

Upstream CPython PR : https://github.com/python/cpython/pull/13346

### PR DESCRIPTION
**Pull request by Hanaasagi**
## What do these changes do?

Compat Python 3.8.

- fix import path, because `asyncio` related exceptions are moved to `asyncio.exceptions` in 3.8.
- fix `make_mocked_coro`
- fix the mock patch, because 3.8 adds empty slots to abstract `asyncio` protocols
- fix some deprecated usage:
    - loop argument is deprecated in `asyncio.sleep`
    - `asyncio.coroutine`
    - `asyncio.open_connection`

## Are there changes in behavior for the user?
I think not.

## Related issue number

It will resolve #3776

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by Hanaasagi** [comment]
This pull request is still work in process. Test with Python 3.8.0b4 in my local environment.

run `make test`

```
9 failed, 2418 passed, 14 skipped, 3 xfailed, 27 error in 156.87s (0:02:36)
```

Most errors are due to Python 3.8 adds empty slots to abstract `asyncio` protocols, please see https://github.com/python/cpython/pull/10889

屏幕快照 2019-09-12 10 04 33

We can not do something like `srv._drain_helper = mock.Mock()` directly.

**Comment by Hanaasagi** [comment]
I don't know  the difference between https://travis-ci.com/Hanaasagi/aiohttp/builds/127773295 and https://travis-ci.com/aio-libs/aiohttp/builds/127770827.

File: aiohttp/web_protocol.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ class RequestHandler(BaseProtocol):
                  '_waiter', '_error_handler', '_task_handler',
                  '_upgrade', '_payload_parser', '_request_parser',
                  '_reading_paused', 'logger', 'access_log',
-                 'access_logger', '_close', '_force_close')
+                 'access_logger', '_close', '_force_close', 'handle_request')
```
Comment (asvetlov) [inline aiohttp/web_protocol.py]: This change looks irrelevant.

Comment (Hanaasagi) [reply aiohttp/web_protocol.py]: There are many tests mock the `handle_request`(not `_handle_request`), but I could not find where call the `handle_request` in aiohttp. Only some docstring mentions it.

https://github.com/aio-libs/aiohttp/blob/dc39183d4351c2fc87dbfa9bcf938567d754571b/aiohttp/web_protocol.py#L97-L99

Is it an old interface?

Comment (asvetlov) [reply aiohttp/web_protocol.py]: Heh, it is an excellent demonstration of mocks weakness.

`handle_request` is `self._manager.request_handler` I guess. 
But our mocked tests didn't fail despite the API was changed.

The fix should be done in a separate PR, this not existing yet pull request is a blocker for your work.

Comment (Hanaasagi) [reply aiohttp/web_protocol.py]: Due to the `asyncio` changes, we could not assign a value to `handle_request`. Actually this block my progress. I need to fix all like https://github.com/aio-libs/aiohttp/blob/dc39183d4351c2fc87dbfa9bcf938567d754571b/tests/test_web_protocol.py#L403

change above to

```Python
with mock.patch.object(web.RequestHandler, 'handle_request', create=True, new=handle):
    pass
```

**Comment by asvetlov** [comment]
Thanks!
